src/blockstore.js

```diff
@@ -39,28 +38,56 @@ function maybeWithSharding (filestore, options) {
  * @returns {Blockstore}
  */
 function createBaseStore (store) {
-  return {
+  /** @type {Blockstore} */
+  const bs = {
     open () {
       return store.open()
     },
 
+    close () {
+      return store.close()
+    },
+
     async * query (query, options) {
-      for await (const { key, value } of store.query(query, options)) {
-        yield new Block(value, keyToCid(key))
+      /** @type {import('interface-datastore').Query} */
```

Comment:
```suggestion
      /** @type {import('interface-blockstore').Query} */
```

Shouldn't be the blockstore query?

Reply:
I merged this, but it's wrong - the blockstore wraps a datastore so you have to convert the blockstore query into a datastore query (e.g. the key types are different)